core/sql/optimizer/RelGrby.h

```diff
@@ -152,6 +154,9 @@ class GroupByAgg : public RelExpr
   inline void setGroupExpr(ValueIdSet &expr) { groupExpr_ = expr;}
   inline void addGroupExpr(ValueIdSet &expr) { groupExpr_ += expr;}
 
+  inline ValueIdList & extraOrderExpr() { return extraOrderExpr_; }
```

Comment:
Yes, it is not a good practice. I need this since this valueID list will be populated (write) by convertToValueIdList(). Maybe I can rename this to extraOrderExprWritable(), So the extraOrderExpr() will not be accidentally written.